llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -142,7 +142,6 @@ bool VPRecipeBase::mayHaveSideEffects() const {
   switch (getVPDefID()) {
   case VPDerivedIVSC:
   case VPPredInstPHISC:
-  case VPScalarCastSC:
   case VPVectorEndPointerSC:
     return false;
   case VPInstructionSC:
```

Comment:
Which now takes care of VPInstructionWithType recipes as well, right?

Reply:
Yep